tests/integration/tier1/convert-offline-systems/prepare_system.py

```diff
@@ -0,0 +1,49 @@
+import re
+import socket
+
+
+# Replace urls in rhsm.conf file to the dogfood satellite server
+def replace_urls_rhsm(satellite_url):
+    with open("/etc/rhsm/rhsm.conf", "r+") as f:
+        file = f.read()
+        # Replacing the urls
+        file = re.sub("hostname = .*", "hostname = {}".format(satellite_url), file)
+        file = re.sub("baseurl = .*", "baseurl = https://{}/pulp/repos".format(satellite_url), file)
+
+        # Setting the position to the top of the page to insert data
+        f.seek(0)
+        f.write(file)
+        f.truncate()
+
+
+# Configure and limit connection to the satellite server only
+def configure_connection(satellite_ip):
+    with open("/etc/dnsmasq.conf", "a") as f:
+        # Satellite url
+        f.write("address=/dogfood.sat.engineering.redhat.com/{}".format(satellite_ip))
+
+        # Everything else is resolved to localhost
+        f.write("address=/#/127.0.0.1")
+
+    with open("/etc/resolv.conf", "w") as f:
+        f.write("nameserver 127.0.0.1")
+
+
+def test_prepare_system(shell):
+    assert shell("yum install dnsmasq wget -y").returncode == 0
+
+    satellite_url = "dogfood.sat.engineering.redhat.com"
```

Comment:
You can de-duplicate the URL string by creating a module-level constant `SATELLITE_URL = "dogfood.sat.engineering.redhat.com"` and using it in the `f.write("address=/dogfood.sat.engineering.redhat.com/{}".format(satellite_ip))` above.

Reply:
Thanks for the idea. I implemented it.